src/sockref.rs

```diff
@@ -20,8 +20,9 @@ use crate::Socket;
 /// the caller must ensure the file descriptor/socket is a valid.
 ///
 /// [`TcpStream`]: std::net::TcpStream
-/// [`AsRawFd`]: std::os::unix::io::AsRawFd
-/// [`AsRawSocket`]: std::os::windows::io::AsRawSocket
+// Don't use intra-doc links because they won't build on every platform.
```

Comment:
What do you mean by this? It should like to docs on rust-lang.org anyway.

Reply:
When the docs are generated on a Unix machine, `AsRawSocket` doesn't exist so Rustdoc isn't able to find the link to it. What std does to have `AsRawSocket` and `AsRawFd` both exist even when documented on Unix (normally only `AsRawFd` would exist because that's the current platform) is very magic, and this magic isn't inherited by the documentation of external crates (so we only get access to one while documenting).